webrecorder/browsertrix
# Configure crawls with JSON

WIP https://github.com/webrecorder/browsertrix-cloud/issues/74

MVP configure crawl with JSON. Properties in the JSON config will override form settings. There is no syncing in between the form and JSON at the moment for simplicity.

### Manual testing
1. Run app and go to Archives > archive > Crawl Templates > new
2. Go to "Pages" section and click "Use JSON editor" toggle. Verify JSON editor shows
3. Interact with JSON editor. Verify JSON is formatted or shows correct error when clicking out of text area.
4. Save config. Verify that invalid JSON is not submitted and valid JSON is formatted into the payload as expected.

### Screenshots
**Toggled off**
<img width="1019" alt="Screen Shot 2022-01-18 at 7 37 27 PM" src="https://user-images.githubusercontent.com/4672952/150059594-134cd299-9d40-4554-a9a3-43f46d252037.png">

**Toggled on**
<img width="723" alt="Screen Shot 2022-01-18 at 7 37 33 PM" src="https://user-images.githubusercontent.com/4672952/150059617-3a5618ff-d585-46f8-a935-8e9bde578011.png">

**Feedback on invalid JSON**
<img width="716" alt="Screen Shot 2022-01-18 at 7 40 16 PM" src="https://user-images.githubusercontent.com/4672952/150059725-501d6cea-f79a-4c9e-b87b-ef5e891a1993.png">
